tests/stests/mtls/mtls_communication.robot

```diff
@@ -0,0 +1,86 @@
+    Given Podman has deleted all existing containers
+    And Ankaios server is started with config "${CONFIGS_DIR}/default.yaml"
+    And Ankaios agent is started with name "agent_B"
+    And all workloads of agent "agent_B" have an initial execution state
+    And Ankaios agent is started with name "agent_A"
+    And all workloads of agent "agent_A" have an initial execution state
+    # Actions
+    When user triggers "ank get workloads"
+    # Asserts
+    Then the workload "nginx" shall have the execution state "Running(Ok)" on agent "agent_A"
+    And the workload "hello1" shall have the execution state "Failed(Lost)" from agent "agent_B"
+    And the workload "hello2" shall have the execution state "Succeeded(Ok)" on agent "agent_B"
+    And the workload "hello3" shall have the execution state "Succeeded(Ok)" on agent "agent_B"
+    [Teardown]    Clean up Ankaios
+
+# [stest->swdd~server-supports-pem-file-paths-as-cli-arguments~]
+# [stest->swdd~agent-supports-pem-file-paths-as-cli-arguments~1]
+# [stest->swdd~cli-supports-pem-file-paths-as-cli-arguments~1]
+Test Ankaios MTLS by providing PEM files via command line arguments
+    [Setup]    Run Keyword    Setup Ankaios without MTLS Setup
+    # Preconditions
+    # This test assumes that all containers in the podman have been created with this test -> clean it up first
+    Given Podman has deleted all existing containers
+    And Ankaios server is started with config "${CONFIGS_DIR}/default.yaml" and PEM files: "${CERTS_DIR}/ca.pem" "${CERTS_DIR}/server.pem" "${CERTS_DIR}/server-key.pem"
+    And Ankaios agent is started with name "agent_B" and PEM files: "${CERTS_DIR}/ca.pem" "${CERTS_DIR}/agent.pem" "${CERTS_DIR}/agent-key.pem"
+    And all workloads of agent "agent_B" have an initial execution state
+    And Ankaios agent is started with name "agent_A" and PEM files: "${CERTS_DIR}/ca.pem" "${CERTS_DIR}/agent.pem" "${CERTS_DIR}/agent-key.pem"
+    And all workloads of agent "agent_A" have an initial execution state
+    # Actions
+    When user triggers "ank --ca_pem ${CERTS_DIR}/ca.pem --crt_pem ${CERTS_DIR}/cli.pem --key_pem ${CERTS_DIR}/cli-key.pem get workloads"
+    # # Asserts
+    Then the workload "nginx" shall have the execution state "Running(Ok)" on agent "agent_A"
+    And the workload "hello1" shall have the execution state "Failed(Lost)" from agent "agent_B"
+    And the workload "hello2" shall have the execution state "Succeeded(Ok)" on agent "agent_B"
+    And the workload "hello3" shall have the execution state "Succeeded(Ok)" on agent "agent_B"
+    [Teardown]    Clean up Ankaios
+
+# [stest->swdd~server-supports-cli-argument-for-insecure-communication~1]
+# [stest->swdd~agent-supports-cli-argument-for-insecure-communication~1]
+# [stest->swdd~cli-supports-cli-argument-for-insecure-communication~1]
+# [stest->swdd~cli-establishes-insecure-communication-based-on-provided-insecure-cli-argument~1]
+Test Ankaios insecure mode by providing --insecure command line arguments
```

Comment:
This stest tests if the cli argument (--insecure) is doing the insecure communication. However, now in the requirement there are also the env variables for the insecure mode explanied but there is not stest testing this. Shall I create a new one or shall I link the "supports insecure argument" reqs just to one other stest or to all other stests since all other stests are using the Insecure mode with env variables.

Reply:
As we are using a lib that automatically handles cli args and env variables, we don't explicitly need to test the env variable.